src/main/java/seedu/realodex/logic/parser/FilterCommandParser.java

```diff
@@ -1,29 +1,89 @@
 package seedu.realodex.logic.parser;
 
 import static seedu.realodex.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.realodex.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.realodex.logic.parser.CliSyntax.PREFIX_REMARK;
+
+import java.util.function.Predicate;
 
 import seedu.realodex.logic.commands.FilterCommand;
 import seedu.realodex.logic.parser.exceptions.ParseException;
-import seedu.realodex.model.person.NameContainsKeyphrasePredicate;
+import seedu.realodex.model.person.Person;
+import seedu.realodex.model.person.predicates.PredicateProducer;
 
 /**
  * Parses input arguments and creates a new FilterCommand object
  */
 public class FilterCommandParser implements Parser<FilterCommand> {
 
+    private static final Prefix[] POSSIBLE_PREFIXES = { PREFIX_NAME, PREFIX_REMARK };
     /**
      * Parses the given {@code String} of arguments in the context of the FilterCommand
      * and returns a FilterCommand object for execution.
+     *
      * @throws ParseException if the user input does not conform the expected format
      */
     public FilterCommand parse(String args) throws ParseException {
-        String keyPhrase = args.trim();
-        if (keyPhrase.isEmpty()) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FilterCommand.MESSAGE_USAGE));
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, POSSIBLE_PREFIXES);
+        PrefixChecker prefixChecker = new PrefixChecker(argMultimap);
+
+        checkEmptyPreamble(prefixChecker);
+        checkOnlyOnePrefixPresent(prefixChecker);
+        checkNoDuplicatePrefix(prefixChecker);
```

Comment:
I was thinking should we move `checkOnlyOnePrefixPresent` and `checkNoDuplicatePrefix` to `PrefixChecker` to delegate all prefix checks to this class and simply `prefixChecker.checkOnlyOnePrefixPresent()` for eg
, its better abstraction imo as FilterCommnadParser should not be checking for `checkOnlyOnePrefixPresent` for example

Reply:
Thanks, actually these methods are in PrefixChecker. This current method is just to abstract out the checking and **throwing of invalid command exception** which imo shld be done within this parser class instead of the PrefixChecker class

Reply:
Makes sense thanks!